src/Compilers/CSharp/Portable/Generated/BoundNodes.xml.Generated.cs

```diff
@@ -10919,72 +10922,72 @@ public NullabilityRewriter(ImmutableDictionary<BoundExpression, (NullabilityInfo
 
         public override BoundNode? VisitDeconstructValuePlaceholder(BoundDeconstructValuePlaceholder node)
         {
-            if (!_updatedNullabilities.TryGetValue(node, out (NullabilityInfo Info, TypeSymbol Type) infoAndType))
+            if (!_updatedNullabilities.TryGetValue(node, out (NullabilityInfo Info, TypeSymbol? Type) infoAndType))
             {
                 return node;
             }
 
-            BoundDeconstructValuePlaceholder updatedNode = node.Update(node.ValEscape, infoAndType.Type);
+            BoundDeconstructValuePlaceholder updatedNode = node.Update(node.ValEscape, infoAndType.Type!);
```

Comment:
How do we know that `infoAndType.Type` isn't null?

Reply:
It corresponds to the type of the bound expression from initial binding. If that type was not null, then the updated type should be not null as well. I've removed `!` for the update cases where `Type` accepts null.

---
In reply to: [463173139](https://github.com/dotnet/roslyn/pull/45640#discussion_r463173139) [](ancestors = 463173139)